project-koku/koku
# [COST-3104] masu api for ocp on cloud processing

## Jira Ticket

[COST-3104](https://issues.redhat.com/browse/COST-3104)

## Description

This adds a Masu API to process OCP on Cloud parquet files. This is done during normal processing, but for a current customer bug we have been incorrectly processing this data. This API and corresponding celery task will delete parquet files for a specific month for OCP on Cloud and reprocess by grabbing the daily data from Trino and using the OCPCloudParquetReportProcessor to create new parquet files. 

## Testing

1. Checkout Branch
2. Load test customer data
3. Using trino check `select sum(lineitem_usageamount) from aws_openshift_daily;` and `select count(*) from aws_openshift_daily;`
4. Hit endpoint http://127.0.0.1:5042/api/cost-management/v1/report/process/openshift_on_cloud/?provider_uuid=0b0d5a9c-4672-4866-a796-f5ebb2b616cd&schema=org1234567&start_date=2022-08-01&end_date=2022-09-30 with dates and provider uuid for your cloud provider that has associated OCP clusters.
5. 3. Using trino check `select sum(lineitem_usageamount) from aws_openshift_daily;` and `select count(*) from aws_openshift_daily;` and verify that data is still correct

## Notes

...
